Panic "invalid memory address or nil pointer dereference" running a query when not connected yet

I like to be able to start my web server even if the database is not accessible. That allows that even if the database is down, the server can still be deployed and wait for the database to be available to resume operations properly instead of failing to start. The problem is that if a request arrives and the database is still not up, a call to `Run` will throw a panic error. 

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x30 pc=0x162a61]

goroutine 16 [running]:
panic(0xf43340, 0xc820014130)
	/usr/local/Cellar/go/1.6/libexec/src/runtime/panic.go:464 +0x3e6
bitbucket.org/jtblin/kigo-api/vendor/github.com/dancannon/gorethink.newQuery(0x1118240, 0x3, 0x1000000000, 0x0, 0x0, 0xc8202a5550, 0x2, 0x2, 0x0, 0x0, ...)
	/Users/jtblin/src/go/src/bitbucket.org/jtblin/kigo-api/vendor/github.com/dancannon/gorethink/utils.go:52 +0x2a1
bitbucket.org/jtblin/kigo-api/vendor/github.com/dancannon/gorethink.(*Session).newQuery(0xc8201e14f0, 0x1118240, 0x3, 0x1000000000, 0x0, 0x0, 0xc8202a5550, 0x2, 0x2, 0x0, ...)
	/Users/jtblin/src/go/src/bitbucket.org/jtblin/kigo-api/vendor/github.com/dancannon/gorethink/session.go:258 +0x97
bitbucket.org/jtblin/kigo-api/vendor/github.com/dancannon/gorethink.Term.Run(0x1118240, 0x3, 0x1000000000, 0x0, 0x0, 0xc8202a5550, 0x2, 0x2, 0x0, 0x0, ...)
	/Users/jtblin/src/go/src/bitbucket.org/jtblin/kigo-api/vendor/github.com/dancannon/gorethink/query.go:238 +0xfe
bitbucket.org/jtblin/kigo-api/db.(*deploymentRepo).Get(0xc820030378, 0xc8202acb08, 0x3, 0x0, 0x0, 0x0)
	/Users/jtblin/src/go/src/bitbucket.org/jtblin/kigo-api/db/deployment.go:59 +0x213
bitbucket.org/jtblin/kigo-api/cmd.(*Server).GetDeployment(0xc8202b0540, 0x1c917e0, 0xc8202fc120, 0xc8202acae0, 0xd, 0x0, 0x0)
	/Users/jtblin/src/go/src/bitbucket.org/jtblin/kigo-api/cmd/grpc.go:92 +0x1c6
bitbucket.org/jtblin/kigo-api/apipb._DeploymentService_GetDeployment_Handler(0x109f940, 0xc8202b0540, 0x1c917e0, 0xc8202fc120, 0xc8202d7680, 0x0, 0x0, 0x0, 0x0)
	/Users/jtblin/src/go/src/bitbucket.org/jtblin/kigo-api/apipb/deployment.pb.go:131 +0x132
bitbucket.org/jtblin/kigo-api/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc820067f00, 0x32403e0, 0xc8202debd0, 0xc8200d7880, 0xc8202a1fe0, 0x1924018, 0xc8202fc0f0, 0x0, 0x0)
	/Users/jtblin/src/go/src/bitbucket.org/jtblin/kigo-api/vendor/google.golang.org/grpc/server.go:497 +0xe13
bitbucket.org/jtblin/kigo-api/vendor/google.golang.org/grpc.(*Server).handleStream(0xc820067f00, 0x32403e0, 0xc8202debd0, 0xc8200d7880, 0xc8202fc0f0)
	/Users/jtblin/src/go/src/bitbucket.org/jtblin/kigo-api/vendor/google.golang.org/grpc/server.go:646 +0x109d
bitbucket.org/jtblin/kigo-api/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc8202515f0, 0xc820067f00, 0x32403e0, 0xc8202debd0, 0xc8200d7880)
	/Users/jtblin/src/go/src/bitbucket.org/jtblin/kigo-api/vendor/google.golang.org/grpc/server.go:323 +0xa0
created by bitbucket.org/jtblin/kigo-api/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
	/Users/jtblin/src/go/src/bitbucket.org/jtblin/kigo-api/vendor/google.golang.org/grpc/server.go:324 +0x9a
```

Rather than having to check on every call if the session is valid, I think it would be preferable to have this check done in the `Run` function of `gorethink` e.g.

```
	if s == nil || !s.IsConnected() {
		return nil, ErrConnectionClosed
	}
```

I could add this check on my side but I'd rather not have to have this check everywhere in my code or have to wrap the `Run` function. I'll send a PR with this, hope that makes sense and you will be OK with it.

--- a/query.go
+++ b/query.go
@@ -242,6 +242,10 @@ func (t Term) Run(s *Session, optArgs ...RunOpts) (*Cursor, error) {
 		opts = optArgs[0].toMap()
 	}
 
+	if s == nil || !s.IsConnected() {
+		return nil, ErrConnectionClosed
+	}
+
 	q, err := s.newQuery(t, opts)
 	if err != nil {
 		return nil, err

--- a/query_test.go
+++ b/query_test.go
@@ -65,3 +65,17 @@ func (s *RethinkSuite) TestQueryRunRawTime(c *test.C) {
 	c.Assert(response["$reql_type$"], test.NotNil)
 	c.Assert(response["$reql_type$"], test.Equals, "TIME")
 }
+
+func (s *RethinkSuite) TestQueryRunNil(c *test.C) {
+	res, err := Expr("Test").Run(nil)
+	c.Assert(res, test.IsNil)
+	c.Assert(err, test.NotNil)
+	c.Assert(err, test.Equals, ErrConnectionClosed)
+}
+
+func (s *RethinkSuite) TestQueryRunNotConnected(c *test.C) {
+	res, err := Expr("Test").Run(&Session{})
+	c.Assert(res, test.IsNil)
+	c.Assert(err, test.NotNil)
+	c.Assert(err, test.Equals, ErrConnectionClosed)
+}

